File: app/src/models/Movie.ts
export default class Movie {
    constructor(public Id: number,
                public Title: string,
                public Year: number,
                public Length: number,
                public TrailerUrl: string | null,
                public ImdbUrl: string | null,
                public ImageUrl: string | null,
                public people: { personId: number, name: string}[],
                public genres: { id: number, name: string }[]) {
                    
                }
}

File: Services/Movies/Movies.Common/Entities/Movie.cs
using System.ComponentModel.DataAnnotations;
using Movies.Common.Constants;

namespace Movies.Common.Entities
{
    public class Movie
    {
        [Required(ErrorMessage = ErrorMessages.ID_REQUIRED)]
        public int Id { get; set; }

        [Required(ErrorMessage = ErrorMessages.TITLE_REQUIRED)]
        public string Title { get; set; }

        [Required(ErrorMessage = ErrorMessages.LENGTH_REQUIRED)]
        public int Length { get; set; }

        [Required(ErrorMessage = ErrorMessages.YEAR_REQUIRED)]
        public uint Year { get; set; }

        public string? TrailerUrl { get; set; }

        public string? ImageUrl { get; set; }
        public string? ImdbUrl { get; set; }
        public List<Genre> Genres { get; set; }
        public List<MoviePerson> People { get; set; }
    }
}
